// Copyright (c) 2020-2021 The Open-Transactions developers
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.

syntax = "proto2";

package opentxs.proto;
option java_package = "org.opentransactions.proto";
option java_outer_classname = "OTEnums";
option optimize_for = LITE_RUNTIME;

enum SignatureRole {
    SIGROLE_ERROR = 0;
    SIGROLE_PUBCREDENTIAL = 1;
    SIGROLE_PRIVCREDENTIAL = 2;
    SIGROLE_NYMIDSOURCE = 3;
    SIGROLE_CLAIM = 4;
    SIGROLE_SERVERCONTRACT = 5;
    SIGROLE_UNITDEFINITION = 6;
    SIGROLE_PEERREQUEST = 7;
    SIGROLE_PEERREPLY = 8;
    SIGROLE_CONTEXT = 9;
    SIGROLE_ACCOUNT = 10;
    SIGROLE_SERVERREQUEST = 11;
    SIGROLE_SERVERREPLY = 12;
}

enum CredentialType {
    CREDTYPE_ERROR = 0;
    CREDTYPE_LEGACY = 1;
    CREDTYPE_HD = 2;
}

enum CredentialRole {
    CREDROLE_ERROR = 0;
    CREDROLE_MASTERKEY = 1;
    CREDROLE_CHILDKEY = 2;
    CREDROLE_CONTACT = 3;
    CREDROLE_VERIFY = 4;
}

enum AsymmetricKeyType {
    AKEYTYPE_ERROR = 0;
    AKEYTYPE_NULL = 1;
    AKEYTYPE_LEGACY = 2;
    AKEYTYPE_SECP256K1 = 3;
    AKEYTYPE_ED25519 = 4;
}

enum SymmetricKeyType {
    SKEYTYPE_ERROR = 0;
    SKEYTYPE_RAW = 1;
    SKEYTYPE_ECDH = 2;
    SKEYTYPE_ARGON2 = 3;
    SKEYTYPE_ARGON2ID = 4;
}

enum SymmetricMode {
    SMODE_ERROR = 0;
    SMODE_CHACHA20POLY1305 = 1;
}

enum KeyMode {
    KEYMODE_ERROR = 0;
    KEYMODE_NULL = 1;
    KEYMODE_PUBLIC = 2;
    KEYMODE_PRIVATE = 3;
}

enum KeyRole {
    KEYROLE_ERROR = 0;
    KEYROLE_AUTH = 1;
    KEYROLE_ENCRYPT = 2;
    KEYROLE_SIGN = 3;
}

enum SourceType {
    SOURCETYPE_ERROR = 0;
    SOURCETYPE_PUBKEY = 1;
    SOURCETYPE_BIP47 = 2;
}

enum SourceProofType {
    SOURCEPROOFTYPE_ERROR = 0;
    SOURCEPROOFTYPE_SELF_SIGNATURE = 1;
    SOURCEPROOFTYPE_SIGNATURE = 2;
}

enum HashType {
    HASHTYPE_ERROR = 0;
    HASHTYPE_NONE = 1;
    HASHTYPE_SHA256 = 2;
    HASHTYPE_SHA512 = 3;
    HASHTYPE_BLAKE2B160 = 4;
    HASHTYPE_BLAKE2B256 = 5;
    HASHTYPE_BLAKE2B512 = 6;
    HASHTYPE_RIPEMD160 = 7;
    HASHTYPE_SHA1 = 8;
    HASHTYPE_SHA256D = 9;
    HASHTYPE_SHA256DC = 10;
    HASHTYPE_BITCOIN = 11;
    HASHTYPE_SIPHASH24 = 12;
}

enum NymMode {
    NYM_ERROR = 0;
    NYM_PRIVATE = 1;
    NYM_PUBLIC = 2;
}

enum AuthorityMode {
    AUTHORITYMODE_ERROR = 0;
    AUTHORITYMODE_INDEX = 1;
    AUTHORITYMODE_FULL = 2;
}

enum SeedType {
    SEEDTYPE_ERROR = 0;
    SEEDTYPE_RAW = 1;
    SEEDTYPE_BIP39 = 2;
    SEEDTYPE_PKT = 3;
}

enum SeedLang {
    SEEDLANG_ERROR = 0;
    SEEDLANG_NONE = 1;
    SEEDLANG_EN = 2;
}
